generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -208,40 +227,37 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
                     : new UIntConstantExpr(maxValue + "u")),
             type, constContext, depth, fuzzer));
     if (isZero) {
+      final BinOp operator = generator.nextBoolean() ? BinOp.SHL : BinOp.SHR;
       return Optional.of(
           new ParenExpr(
               new BinaryExpr(
                   makeOpaqueZero(type, constContext, depth, fuzzer),
-                  applyIdentityFunction(shiftValue, type, constContext, depth, fuzzer),
+                  shiftValue,
                   operator)));
     } else {
-      assert !isZero;
-      assert operator == BinOp.SHL || operator == BinOp.SHR;
       // We're going to shift twice in opposite directions by the same value.
```

Comment:
How about here you do:

final Expr shiftBackValue = generaor.nextBool() ? shiftValue.clone() : (another fuzzed expression clamped to maxValue);

Then below change the first shiftValue.clone() to shiftValue, and the second shiftValue.clone() to shiftBackValue.

This will be nice as it will sometimes lead to some duplicated expressions (giving the compiler room for common subexpression elimination optimizations), and other times leads to very different but equivalent expressions (testing the compiler in other ways).

As you'll be generating the big clamped expression twice, it would be worth factoring that out into a private utility method.

Reply:
Done.